kusto/query_builder.go

```diff
@@ -506,6 +507,143 @@ func NewStmt(query stringConstant, options ...StmtOption) Stmt {
 	return s
 }
 
+func (s Stmt) AddDatabase(query string) Stmt {
+	return s.SafeAddName(query)
+}
+
+func (s Stmt) AddTable(query string) Stmt {
+	return s.SafeAddName(query)
+}
+
+func (s Stmt) AddColumn(query string) Stmt {
+	return s.SafeAddName(query)
+}
+
+func (s Stmt) AddFunction(query string) Stmt {
+	return s.SafeAddName(query)
+}
+
+func (s Stmt) SafeAddName(query string) Stmt {
+	if !(strings.Contains(query, "'")) {
+		query = "['" + query + "']"
+	}
+
+	if !(query == "" || strings.HasPrefix(query, "[")) {
+		query = "[\"" + query + "\"]"
+	}
+
+	s.queryStr = s.queryStr + query
+	return s
+}
+
+func (s Stmt) AddQuotedString(value string, hidden bool) Stmt {
+	if value == "" {
```

Comment:
This was a null check.
I believe for an empty string the correct output should be a quoted empty string i.e.
```
""
```

Reply:
If i understood correctly, this is the way to do a null check on string in Go (cannot directly check if it's nil). should we keep it like this then? im returning s because the actual effect of it would be no change to the statement that called this function. (and we are "returning" a stmt, so i cant return  an empty string)